packages/blocks/src/api/parser/index.js

```diff
@@ -50,6 +50,11 @@ import { applyBuiltInValidationFixes } from './apply-built-in-validation-fixes';
  * @property {WPRawBlock} [__unstableBlockSource] Un-processed original copy of block if created through parser.
  */
 
+/**
+ * @typedef  {Object}  ParseOptions
+ * @property {boolean} __unstableSkipMigrationLogs If a block is migrated from a deprecated version, skip logging the migration details.
```

Comment:
the more I think about this the less I like polluting the interface with very specific behavioral changes for calling the parser, especially since the option here is to avoid a nuisance instead of effecting actual change in behavior.

❓ what have been our explorations for using `debug()` (or a variant thereof) to enable opt-in logging for different subsystems in the project?

Gutenberg is _noisy_ and that has caused me grief not only while developing for the project itself (these unexpected migration logs for blocks that don't exist within a post being one particular annoying case) but also while integrating the project into other applications and when testing.

The greedy use of `console` can interfere in a lot of unexpected ways.

❓ How valuable are these migration logs in practice?

Do we have evidence that people use them or get value from them? There's a hint of "crying wolf" with how they appear on every valid page load that makes them far less valuable IMO. Sure, we could add a new special-case for patterns, but then someone will want to know if the patterns need updating or someone else will want to special-case another situation, then we'll have conflicting flags that govern the console use etc…

Would anyone object to outright removing the logs entirely and seeing if anyone cares?

Reply:
>what have been our explorations for using debug() (or a variant thereof) to enable opt-in logging for different subsystems in the project?

What do you mean with `debug`? 

>How valuable are these migration logs in practice?

Personally I don't have an informed opinion about this, but I don't see much value in the patterns case(this PR). In the case of the actual content, I see no harm being there as it's not something a user could see often..

Regarding the case of someone wanting to check the patterns, it should be the pattern authors responsibility and is more convoluted as I [explain here](https://github.com/WordPress/gutenberg/pull/39404#discussion_r825722659).

Reply:
> Going back to this PR, even with nuanced needs for patterns, I'm not sure those are actually unique in a way that other needs aren't and I'm really doubting it warrants a special-casing parameter to the parser. That makes me think we can accomplish this with a more general and simpler approach: either eliminate all the logs or make these logs opt-in so that people can make a decision within their browser whether or not to show them.
> 
> Even a `NODE_ENV === 'development'` check could potentially suffice.

"crying wolf" is a real issue, but the solution should never come from gating notices behind `NODE_ENV`, which is specific to developing Gutenberg itself and is far removed from the realities of WordPress extenders working on patterns, themes, etc., as well as those building bespoke sites. At best, there could be a case for gating behind `WP_DEBUG`, but personally I'm not convinced that that's enough.

We should proceed with this PR, which alone will greatly reduce the console noise, thus freeing up the console for errors over which the user/developer actually has some agency.

Reply:
just curious, why not go ahead and gate it behind `WP_DEBUG`? I don't think `NODE_ENV` is important as the mechanism for quieting the console, but seems like some flag to opt-in to these messages would do great good for the general case where we get hundreds of console logs for unexplained reasons.